packages/service-library/src/servicelib/archiving_utils.py

```diff
@@ -82,12 +98,16 @@ def _zipfile_single_file_extract_worker(
         if is_dir:
             destination_path.mkdir(parents=True, exist_ok=True)
             return destination_path
-
-        with zf.open(name=file_in_archive) as zip_fp:
-            with open(destination_path, "wb") as destination_fp:
-                for chunk in _read_in_chunks(zip_fp):
-                    destination_fp.write(chunk)
-                return destination_path
+        desc = f"decompressing {zip_file_path}:{file_in_archive.filename} -> {destination_path}\n"
+        with zf.open(name=file_in_archive) as zip_fp, destination_path.open(
```

Comment:
Q: is this equivalent to three embedded ``with``-statements, i.e.
```python
    with ctx1 as a:
        with ctx2 as b:
             with ctx3 as c:
                 ...
```
probably i deserve here a RTFM answer ... :-)

Reply:
yes!